### PR TITLE
Adds retry after header

### PR DESCRIPTION
**Pull request by anglinb**
Following https://developer.apple.com/documentation/appstoreserverapi/identifying_rate_limits, I added the retry after to the error so the client can respect the rate limit correctly.

### CODE REVIEW

File: src/Errors.ts [hunk 1]
```diff
@@ -13,15 +13,17 @@ export class AppStoreError extends Error {
 
   errorCode: number
   isRetryable: boolean
+  retryAfter?: number
 
   // https://developer.apple.com/documentation/appstoreserverapi/ratelimitexceedederror
   isRateLimitExceeded: boolean
 
-  constructor(errorCode: number, errorMessage: string) {
+  constructor(errorCode: number, errorMessage: string, { retryAfter}:{ retryAfter?: number } = {}) {
```
Comment (agisboye) [inline src/Errors.ts:21]: The last parameter looks a little clunky. Why not just use `retryAfter?: number`?
```typescript
constructor(errorCode: number, errorMessage: string, retryAfter?: number) {...}
```

**Review by agisboye — CHANGES_REQUESTED** [review]
Thanks for the PR! Would you mind taking a look at the comment before I merge this?

File: src/Errors.ts (hunk 1 above)
Comment (anglinb) [reply src/Errors.ts]: Sure!

**Comment by anglinb** [comment]
Ok tweaked the constructor. Made this change a while ago in my fork, can't remember why I decided to add the object param instead of just an optional number.  🤷 Either way I think it's better your way!

**Comment by agisboye** [comment]
Thank you. I've merged and released v0.15.3.

**Pull request merged**